src/gtk/webview_webkit2.cpp

```diff
@@ -1092,14 +1097,107 @@ wxString wxWebViewWebKit::GetPageText() const
     }
     return wxString();
 }
+static void wxgtk_run_javascript_cb(WebKitWebView *,
+                                    GAsyncResult *res,
+                                    GAsyncResult **res_out)
+{
+    *res_out = (GAsyncResult*)g_object_ref(res);
+}
+
+bool JSResultToString(GObject *object, GAsyncResult *result, wxString* output)
+{
+    WebKitJavascriptResult  *js_result;
+    JSValueRef              value;
+    JSGlobalContextRef      context;
+    wxGtkError              error;
+
+    js_result = webkit_web_view_run_javascript_finish(WEBKIT_WEB_VIEW (object),
+        (GAsyncResult *)result, error.Out());
+
+    if ( !js_result )
+    {
+        wxLogWarning(_("Error running Javascript: %s"), error.GetMessage());
+        return false;
+    }
+
+    context = webkit_javascript_result_get_global_context (js_result);
+    value = webkit_javascript_result_get_value (js_result);
+
+
+    JSStringRef js_value;
+    gsize length;
+    JSValueRef exception = NULL;
+
+    js_value = (JSValueIsObject(context, value)) ?
+               JSValueCreateJSONString(context, value, 0, &exception) :
+               JSValueToStringCopy (context, value, &exception);
+
+    if ( exception )
+    {
+        JSStringRef ex_value = JSValueToStringCopy(context, exception, NULL);
+        gsize ex_length;
+        ex_length = JSStringGetMaximumUTF8CStringSize (ex_value);
+        wxGtkString str(g_new(gchar, ex_length));
+
+        JSStringGetUTF8CString (ex_value, (gchar*)str.c_str(), ex_length);
+        JSStringRelease (ex_value);
+
+        wxLogWarning(_("Exception running Javascript: %s"), wxString::FromUTF8(str));
+
+        webkit_javascript_result_unref (js_result);
```

Comment:
Using manual unref calls is error-prone and unnecessarily verbose, you must define a RAII wrapper class similar to `wxGtkObject` for `WebKitJavascriptResult` instead.